nightfall-optimist/src/services/transaction-sign-send.mjs

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-param-reassign */
+
+import config from 'config';
+import { web3 } from '@polygon-nightfall/common-files/utils/contract.mjs';
+import { estimateGas, estimateGasPrice } from '@polygon-nightfall/common-files/utils/gas.mjs';
+import logger from '@polygon-nightfall/common-files/utils/logger.mjs';
+import { Mutex } from 'async-mutex';
+
+const { GAS, GAS_MULTIPLIER, GAS_ESTIMATE_ENDPOINT, GAS_PRICE, GAS_PRICE_MULTIPLIER } = config;
+
+const nonceMutex = new Mutex();
+
+const isWeb3Listening = async () => {
+  try {
+    const isListening = await web3.eth.net.isListening();
+    return isListening;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+};
+
+export async function getAddressNonce(ethAddress) {
+  try {
+    const nonce = await web3.eth.getTransactionCount(ethAddress);
+    return nonce;
+  } catch (error) {
+    throw new Error(error.message);
+  }
+}
+
+// TODO document
+export async function createSignedTransaction(nonce, ethPrivateKey, from, to, data, value = 0) {
+  // Check if web3 ws is available
+  const result = await isWeb3Listening();
+  if (result) {
```

Comment:
I'd invert this `if` so that we return the negative asap since it is just a piece of code, while the code inside the `if` statement is pretty big. e.g. 
`if(! result) {
    throw new Error('Web3 ws not listening, try again later');
}
... // the rest...

Reply:
Of course